lib/createStylelint.js

```diff
@@ -16,43 +16,29 @@ const STOP_DIR = IS_TEST ? process.cwd() : undefined;
 /**
  * The stylelint "internal API" is passed among functions
  * so that methods on a stylelint instance can invoke
- * each other while sharing options and caches
+ * each other while sharing options and caches.
+ *
  * @param {import('stylelint').StylelintStandaloneOptions} options
  * @returns {StylelintInternalApi}
  */
 module.exports = function createStylelint(options = {}) {
-	/** @type {Partial<StylelintInternalApi>} */
+	/** @type {StylelintInternalApi} */
+	// @ts-expect-error -- TS2740: Type '{ _options: StylelintStandaloneOptions; }' is missing the following properties from type 'StylelintInternalApi'
```

Comment:
[note] I think we can allow the TS2740 error to reduce type annotations of `StylelintInternalApi`.